### PR TITLE
(cleanup, nobug) Start BVC split work, splitting out Nav Delegate and KeyCommands

### PR DESCRIPTION
**Pull request by garvankeeley**
These were low-hanging fruit items in regards to splitting up BrowserViewController.swift. There is much more to go.

ASIDE: I am also fixing all SwiftLint warnings in bvc

### CODE REVIEW

**Comment by garvankeeley** [comment]
Ugh, it shows BVC.swift as having fully changed in the diff view because I moved it into a new folder BrowserViewController/
I'll move it out and see if the diff looks better. We can move the file later, more important now to have a cleaner diff.

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
BrowserViewController.swift | 1474 | Arguments can be omitted when matching enums with associated types if they are not used. 
BrowserViewController.swift | 1476 | Arguments can be omitted when matching enums with associated types if they are not used. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -484,6 +484,15 @@ class BrowserViewController: UIViewController {
         log.debug("BVC done.")
     }
 
+    override var canBecomeFirstResponder: Bool {
```
Comment (garvankeeley) [inline Client/Frontend/Browser/BrowserViewController.swift]: If this is only for key command handling, it could go into that extension

**Comment by garvankeeley** [comment]
BTW, I'll fix that swiftlint error reported above

**Pull request merged**